scylla/src/transport/iterator.rs

```diff
@@ -689,7 +689,7 @@ where
                     .load_balancing_policy
                     .on_query_success(&self.statement_info, elapsed, node);
 
-                self.paging_state = rows.paging_state.take();
```

Comment:
Here as well (regarding @Lorak-mmk 's comment). I think this should belong to other commit. I am not even sure why we were modifying received `rows` (by moving the paging state out of them), even though we send them via some channel later.

Reply:
We can do that because the iterator manages paging state itself, and it does not expose any getters of it.